### ISSUE TITLE
Mac OS SDL2 compile failing

### ISSUE THREAD

**Issue by jonesmz** [TheOpenSpaceProgram/osp-magnum#173]
Probably there's a CMake level config option that can be used.

```
 /Users/runner/work/osp-magnum/osp-magnum/3rdparty/SDL2/src/joystick/iphoneos/SDL_mfijoystick.m:1169:16: error: implicit declaration of function 'typeof' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        __weak typeof(self) weakSelf = self;
               ^
/Users/runner/work/osp-magnum/osp-magnum/3rdparty/SDL2/src/joystick/iphoneos/SDL_mfijoystick.m:1169:16: error: expected identifier or '('
/Users/runner/work/osp-magnum/osp-magnum/3rdparty/SDL2/src/joystick/iphoneos/SDL_mfijoystick.m:1171:38: error: use of undeclared identifier 'weakSelf'
            SDL_RumbleMotor *_this = weakSelf;
                                     ^
/Users/runner/work/osp-magnum/osp-magnum/3rdparty/SDL2/src/joystick/iphoneos/SDL_mfijoystick.m:1180:38: error: use of undeclared identifier 'weakSelf'
            SDL_RumbleMotor *_this = weakSelf;
                                     ^
/Users/runner/work/osp-magnum/osp-magnum/3rdparty/SDL2/src/joystick/iphoneos/SDL_mfijoystick.m:1471:24: warning: 'controllerPausedHandler' is deprecated: first deprecated in macOS 10.15 - controllerPausedHandler has been deprecated. Use the Menu button found on the controller's profile, if it exists. [-Wdeprecated-declarations]
            controller.controllerPausedHandler = nil;
```

**Comment by Capital-Asterisk** [comment]
I mentioned this in the discord, but I'm writing this here for good measure. This wouldn't be an error if it wasn't for `-Werror`, which makes me suspect that SDL2 needs a certain compiler version. I'd suggest we switch to using the official precompiled binaries instead of making this our problem, which can be automatically downloaded.

**Comment by jonesmz** [comment]
SDL Github folks asked if it's cause we don't allow compiler extensions at the top level.
I think they're going to fix their code for us in their next release.

**Comment by Capital-Asterisk** [comment]
Fixed in #177
